### PR TITLE
Making PRs is fun

### PR DESCRIPTION
**Pull request by AdyashaDash**
Please merge your own commits to master?

### CODE REVIEW

File: Task2/T2-Pat/run.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+
+import sys
```
Comment (zhaop) [inline Task2/T2-Pat/run.py]: import 👩

File: Task2/T2-Pat/run.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import time
```
Comment (zhaop) [inline Task2/T2-Pat/run.py]: import 🕙

File: Task2/T2-Pat/run.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import matplotlib.pyplot as plt
+import tensorflow.contrib.learn as learn
+from tensorflow.contrib.layers import real_valued_column
+from tensorflow import train, logging
```
Comment (zhaop) [inline Task2/T2-Pat/run.py]: from tensorflow import 🚄

File: Task2/T2-Pat/run.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    if idx_end > len(all_params):
+        idx_end = len(all_params)
+        print('Note: Clamping interval end to last few param sets')
+
+    print('Processing [{:d}, {:d})'.format(idx_start, idx_end))
+
+    for params in all_params[idx_start:idx_end]:
+    
+        steps = params[0]
```
Comment (zhaop) [inline Task2/T2-Pat/run.py]: 👣 = params[0]

File: Task1/T1-Pat/run.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
```
Comment (zhaop) [inline Task1/T1-Pat/run.py]: #!/usr/bin/env 🐍3

File: Task2/T2-Pat/dropout3/run.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+            X_subverify = X_train[verify_idx, :]
+            Y_subverify = Y_train[verify_idx]
+
+            result = validation(model, X_subtrain, Y_subtrain, X_subverify, Y_subverify, steps=steps)
+            accuracies = np.append(accuracies, result['accuracy'])
+            losses = np.append(losses, result['loss'])
+
+            print("{:.2f} ".format(result['accuracy']), end='')
+            sys.stdout.flush()
+        timing += time.time()
```
Comment (zhaop) [inline Task2/T2-Pat/dropout3/run.py]: ⏱ += 🕙.🕙()

File: Task2/T2-Pat/dropout3/run.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        print("{:.1f}s".format(timing), end='\t')
+
+        # Run on test set
+        timing = -time.time()
+        Y_predict = model.predict(X_test)
+        timing += time.time()
+        print("{:.1f}s".format(timing), end='')
+
+        print('')
```
Comment (zhaop) [inline Task2/T2-Pat/dropout3/run.py]: 🖨('')

File: Task2/T2-Pat/etc/etc.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+
+import sys
+import time
+import matplotlib.pyplot as plt
+from sklearn.ensemble import ExtraTreesClassifier, RandomForestClassifier
```
Comment (zhaop) [inline Task2/T2-Pat/etc/etc.py]: from sk👩‍🏫.ensemble import Extra🌳🌳Classifier, Random🌳🌲🌳🌲Classifier

**Comment by zhaop** [comment]
💯 :shipit:

**Comment by zhaop** [comment]
@AdyashaDash 💯 :shipit:

**Comment by AdyashaDash** [comment]
Such a piece of art this is @zhaop 

Check it out @metamercurial @simonmaurer

**Pull request merged**